internal/pkg/otel/configtranslate/otelconfig.go

```diff
@@ -0,0 +1,366 @@
+			"pipelines": pipelinesConfig,
+		},
+	}
+	return confmap.NewFromStringMap(fullConfig), nil
+}
+
+// getReceiversConfigForComponent returns the receivers configuration for a component. Usually this will be a single
+// receiver, but in principle it could be more.
+func getReceiversConfigForComponent(comp *component.Component, info info.Agent, outputQueueConfig map[string]any) (map[string]any, error) {
+	receiverType, err := getReceiverTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+	// this is necessary to convert policy config format to beat config format
+	defaultDataStreamType, err := getDefaultDatastreamTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+
+	// get inputs for all the units
+	// we run a single receiver for each component to mirror what beats processes do
+	var inputs []map[string]any
+	for _, unit := range comp.Units {
+		if unit.Type == client.UnitTypeInput {
+			unitInputs, err := getInputsForUnit(unit, info, defaultDataStreamType)
+			if err != nil {
+				return nil, err
+			}
+			inputs = append(inputs, unitInputs...)
+		}
+	}
+
+	receiverId := getReceiverID(receiverType, comp.ID)
+	// Beat config inside a beat receiver is nested under an additional key. Not sure if this simple translation is
+	// always safe. We should either ensure this is always the case, or have an explicit mapping.
+	beatName := strings.TrimSuffix(receiverType.String(), "receiver")
+	beatDataPath := filepath.Join(paths.Home(), "run", comp.ID)
+	receiverConfig := map[string]any{
+		beatName: map[string]any{
+			"inputs": inputs,
+		},
+		// the output needs to be otelconsumer
+		"output": map[string]any{
+			"otelconsumer": map[string]any{},
+		},
+		// just like we do for beats processes, each receiver needs its own data path
+		"path": map[string]any{
+			"data": beatDataPath,
+		},
+	}
+	// add the output queue config if present
+	if outputQueueConfig != nil {
+		receiverConfig["output"] = outputQueueConfig
+	}
+	return map[string]any{
+		receiverId.String(): receiverConfig,
+	}, nil
+}
+
+// getReceiversConfigForComponent returns the exporters configuration for a component. Usually this will be a single
+// exporter, but in principle it could be more.
+func getExportersConfigForComponent(comp *component.Component) (map[string]any, error) {
+	exportersConfig := map[string]any{}
+	exporterType, err := getExporterTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+	for _, unit := range comp.Units {
+		if unit.Type == client.UnitTypeOutput {
+			unitExportersConfig, expErr := unitToExporterConfig(unit, exporterType)
+			if expErr != nil {
+				return nil, expErr
+			}
+			for k, v := range unitExportersConfig {
+				exportersConfig[k] = v
+			}
+		}
+	}
+	return exportersConfig, nil
+}
+
+// getBeatNameForComponent returns the beat binary name that would be used to run this component.
+func getBeatNameForComponent(comp *component.Component) string {
+	// TODO: Add this information directly to the spec?
```

Comment:
Is this TODO still relevant? Maybe is it worth opening a follow-up issue for it ?

Reply:
Yeah, that's the plan.

Reply:
Filed an issue here: https://github.com/elastic/ingest-dev/issues/5129.